src/test/java/org/takes/tk/TkProxyTest.java

```diff
@@ -30,26 +30,140 @@
 import org.junit.Test;
 import org.takes.http.FtRemote;
 import org.takes.rq.RqFake;
+import org.takes.rq.RqWithHeaders;
 import org.takes.rs.RsPrint;
 
 /**
  * Test case for {@link TkProxy}.
+ * The test verify the different HTTP methods (GET, POST, etc),
+ * as well as the different combinations of request/response headers.
  * @author Dragan Bozanovic (bozanovicdr@gmail.com)
  * @version $Id$
  * @since 0.25
- * @todo #377:30min/DEV We need more tests for TkProxy.
- *  The tests should verify the different HTTP methods (GET, POST, etc),
- *  as well as the different combinations of request/response headers.
  */
 public final class TkProxyTest {
 
     /**
-     * TkProxy can work.
+     * Test string.
+     */
+    private static final String HELLO_WORLD = "hello, world!";
+
+    /**
+     * Format string.
+     */
+    private static final String FORMAT = "%s:%d";
+
+    /**
+     * Dash constant.
+     */
+    private static final String DASH = "/";
+
+    /**
+     * Response string.
+     */
+    private static final String HEADER = "X-Takes-TkProxy:";
+
+    /**
+     * OK response.
+     */
+    private static final String OK = "HTTP/1.1 200 OK";
+
+    /**
+     * TkProxy can work with different HTTP methods
+     * (GET, POST, OPTIONS, PUT, DELETE).
+     * Plus 4 random headers checked.
+     * TkProxy can throw IOException on invalid HTTP methods.
      * @throws Exception If some problem inside
      */
     @Test
-    public void justWorks() throws Exception {
-        new FtRemote(new TkFixed("hello, world!")).exec(
+    public void actsOnHttpMethodsWithRequestHeaders() throws Exception {
```

Comment:
@aschworer Too many methods should be suppressed here. It's obvious, that number of tests could be more that 10 here